atom/browser/api/atom_api_web_contents.cc

```diff
@@ -789,9 +800,15 @@ void WebContents::AddNewContents(content::WebContents* source,
     if (disposition != WindowOpenDisposition::NEW_WINDOW &&
         disposition != WindowOpenDisposition::NEW_POPUP) {
       if (!browser) {
-        // TODO(bridiver) - this should be the source owner_window
-        browser = owner_window()->browser();
-        tab_helper->SetWindowId(browser->session_id().id());
+        auto relay = atom::NativeWindowRelay::FromWebContents(source);
+        if (relay) {
+          owner_window = relay->window.get();
+          browser = owner_window->browser();
+          tab_helper->SetWindowId(browser->session_id().id());
+        } else {
+          DeleteWebContents(new_contents);
```

Comment:
Use `DestroyWebContents` may be?

Reply:
how does destroying the webcontents fix the race condition?